pandas/rpy/common.py

```diff
@@ -134,35 +137,47 @@ def _convert_str_vector(obj):
     return arr
 
 
-def _convert_DataFrame(rdf):
+def _convert_factor(obj, **kwargs):
+    if kwargs.get("factors_as_strings", True):
+        levels = np.asarray(obj.levels)
+        values = np.asarray(obj)
+        if com.is_float_dtype(values):
+            mask = np.isnan(values)
```

Comment:
@jseabold: I don't see a test exercising this code path (when `is_float_dtype(values)` is `True`). Can you explain its purpose and/or perhaps add a test?